### PR TITLE
Progbar update

### LINKED ISSUE #338: progsteps only reports for some of the simulation
Progsteps reporting on simulations always seems to stall around the 30-36% mark. This has happened for every simulation I've run. It will magically tick to 100% at the end though. See sample output attached.

@kbharatgehlot has reported similar experiences in his sims ever since #313 was implemented.
Possibly related to #336 if the counter update changes the behavior.

<details> <summary>pyuvsim output</summary>

```

(py38_sim) mkolopanis@excalibur [15:17:47] [~/MWA_simpleds/simulation]                                                                                                                                              
-> % /home/mkolopanis/src/anaconda/envs/py38_sim/bin/mpiexec --hostfile mpi_hosts --map-by socket:PE=4 --oversubscribe -n 30 /home/mkolopanis/src/anaconda/envs/py38_sim/bin/python  /home/mkolopanis/src/anaconda/e
nvs/py38_sim/bin/run_param_pyuvsim.py /data3/MWA/MWA_simpleds/simulation/obsparam_ateam_band1.yaml                                                                                                                  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]  
Sky Model setup took 0.2887161999999144 min                                                                                                                                                                         

Nbls: 9                                                                                                                                                                                                             

Ntimes: 117                                                                                                                                                                                                         

Nfreqs: 256                                                                                                                                                                                                         

Nsrcs: 8145                                                                                                                                                                                                         

Tasks:  269568.0                                                                                                                                                                                                    

1.00% completed. 0:00:22.280570  elapsed. 0:36:43.860488 remaining.                                                                                                                                                 
                                                                                                                                                                                                                    
2.00% completed. 0:00:43.022723  elapsed. 0:35:05.865457 remaining.                                                                                                                                                 
                                                                                                                                                                                                                    
3.00% completed. 0:01:02.608144  elapsed. 0:33:42.020336 remaining.                                                                                                                                                 
                                                                   
4.00% completed. 0:01:21.508201  elapsed. 0:32:33.878808 remaining.                                                                                                                                        
                                                                                                                                                                                                                    
5.00% completed. 0:01:40.331705  elapsed. 0:31:44.428293 remaining.                                                                                                                                                 
                                                                                                                                                                                                                    
6.01% completed. 0:01:58.451812  elapsed. 0:30:53.438391 remaining.                                                                                                                                                 
                                                                                                                                                                                                                    
7.01% completed. 0:02:17.158609  elapsed. 0:30:19.943445 remaining.                                                                                                                                                 
                                                                                                                                                                                                                    
8.01% completed. 0:02:36.320631  elapsed. 0:29:54.922180 remaining.                                                                                                                                                 
                                                                                                                                                                                                                    
9.01% completed. 0:02:56.126629  elapsed. 0:29:38.026450 remaining.                                                                                                                                                 
                                                                                                                                                                                                                    
10.01% completed. 0:03:13.468964  elapsed. 0:28:58.482942 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
11.02% completed. 0:03:31.110682  elapsed. 0:28:25.071153 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
12.02% completed. 0:03:51.710322  elapsed. 0:28:15.941074 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
13.02% completed. 0:04:10.088111  elapsed. 0:27:50.315019 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
14.02% completed. 0:04:29.664694  elapsed. 0:27:33.378934 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
15.02% completed. 0:04:47.209566  elapsed. 0:27:04.457305 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
16.03% completed. 0:05:07.600854  elapsed. 0:26:51.384264 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
17.04% completed. 0:05:26.268100  elapsed. 0:26:28.800472 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
18.04% completed. 0:05:45.207819  elapsed. 0:26:08.363666 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
19.04% completed. 0:06:04.299522  elapsed. 0:25:48.954328 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
20.04% completed. 0:06:24.412977  elapsed. 0:25:33.653089 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
21.04% completed. 0:06:42.514672  elapsed. 0:25:10.345406 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
22.04% completed. 0:07:02.636834  elapsed. 0:24:54.559061 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
23.05% completed. 0:07:19.125984  elapsed. 0:24:26.326230 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
24.05% completed. 0:07:38.982396  elapsed. 0:24:09.559693 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
25.05% completed. 0:07:56.516800  elapsed. 0:23:45.767960 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
26.05% completed. 0:08:15.944371  elapsed. 0:23:27.695609 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
27.05% completed. 0:08:34.517706  elapsed. 0:23:07.376119 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
28.06% completed. 0:08:52.028901  elapsed. 0:22:44.306722 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
29.06% completed. 0:09:10.972078  elapsed. 0:22:25.092685 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
30.06% completed. 0:09:27.664865  elapsed. 0:22:00.777754 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
31.06% completed. 0:09:46.915355  elapsed. 0:21:42.676046 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
32.06% completed. 0:10:04.412510  elapsed. 0:21:20.612728 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
33.07% completed. 0:10:21.822834  elapsed. 0:20:58.671879 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
34.07% completed. 0:10:40.753049  elapsed. 0:20:40.061859 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
35.07% completed. 0:10:59.882858  elapsed. 0:20:21.703601 remaining.                                                                                                                                                
                                                                                                                                                                                                                    
100.00% completed. 0:29:51.381791  elapsed. 0:00:00 remaining.                                                                                                                                                      
                                                                                                                                                                                                                    
Calculations Complete. 
```
</details>

### PR DESCRIPTION
**Pull request by mkolopanis**
Keep writing progsteps to terminal after rank 0 finishes work
## Description

Changed to a non-blocking barrier after the work execution loop.
Registered a sys.stdout.close for non rank 0 processes. Newer openmpi versions complain about an open file handler.
I could probably add a time.sleep() to the end of the loop for updating progess too, otherwise the rank 0 will just be writing a ton. maybe a few seconds?

## Motivation and Context

closes #338 
This only resolves the actual issue of the progsteps stopping writing. It does **not** address the delay we were investigating in that issue. I think we should probably make the delay thing a new issue anyway.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Reference simulation update or replacement
- [ ] Documentation change (documentation changes only)
- [ ] Version change
- [ ] Build or continuous integration change

## Checklist:

For all pull requests:
- [x] I have read the [contribution guide](CONTRIBUTING.md).
- [x] My code follows the code style of this project.

Bug fix checklist:
- [ ] My fix includes a new test that breaks as a result of the bug (if possible).
- [x] All new and existing tests pass.
- [ ] I have checked that I reproduce the reference simulations or if there are differences they are explained below (if appropriate). If there are changes that are correct, I will update the reference simulation files after this PR is merged.
- [ ] I have checked (e.g., using the benchmarking tools) that this change does not significantly increase typical runtimes. If it does, I have included a justification in the comments on this PR.
- [x] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvsim/blob/master/CHANGELOG.md).

### CODE REVIEW

File: pyuvsim/mpi.py [hunk 1]
```diff
@@ -67,6 +67,7 @@ def start_mpi(block_nonroot_stdout=True):
         # For non-root ranks, do not print to stdout.
         # (Uncovered until we have multi-rank tests)
```
Comment (aelanman) [inline pyuvsim/mpi.py:68]: Could you remove this parenthetical from the comment? These lines are probably covered now...

**Review by aelanman — COMMENTED** [review]
This looks good! Thanks for putting this together. There's one change you made that I'm pretty sure isn't necessary, but otherwise this should be ready to go.

I agree that we should make a separate issue for the nonzero-rank delay, although I've been having some trouble lately in reproducing it.

File: pyuvsim/uvsim.py [hunk 2]
```diff
@@ -482,7 +482,16 @@ def run_uvdata_uvsim(input_uv, beam_list, beam_dict=None, catalog=None, quiet=Fa
...
         if rank == 0 and not quiet:
             pbar.update(cval)
-    comm.Barrier()
+
+    request = comm.Ibarrier()
+
+    current_value = count.current_value()
+    while not request.Test():
+        cval = count.current_value()
+        if rank == 0 and not quiet and cval != current_value:
```
[the remark this answers is not in the document]
Comment (mkolopanis) [reply pyuvsim/uvsim.py:491]: ah great! thanks for letting me know.

**Comment by mkolopanis** [comment]
@aelanman what are your thoughts about adding a `time.sleep` to the end of the updating loop?

**Comment by aelanman** [comment]
It shouldn't be necessary for the progsteps, because it only prints when the counter value has stepped by 1% of the total number of tasks (or for every step if there are fewer than 100 tasks).
However, it would reduce the number of calls to `counter.current_value()`... I'm not sure if having rank 0 constantly locking and unlocking the counter would slow down other processes.

**Comment by mkolopanis** [comment]
Are we interested in print statements to show time to complete some parts like these?
image

**Comment by aelanman** [comment]
That looks like a good idea. You should probably include `" min"` at the end of the format strings so the user knows the time is measured in minutes.

**Comment by mkolopanis** [comment]
Time is an astropy object (or subclass if from the moon repo right?) it should print out the units too.

**Comment by aelanman** [comment]
Oh right, of course. Looks good!

**Comment by mkolopanis** [comment]
@aelanman I think this is ready for another look.

**Review by aelanman — APPROVED** [review]
Looks good! Thanks @mkolopanis

**Pull request merged**